app/assets/javascripts/wpcc/components/StepTransition.js

```diff
@@ -0,0 +1,165 @@
+  'use strict';
+
+  var StepTransition,
+    defaultConfig = {
+      activeFormClass: 'form--active',
+      inactiveFormClass: 'form--inactive',
+    };
+
+  StepTransition = function($el, config) {
+    StepTransition.baseConstructor.call(this, $el, config, defaultConfig);
+
+    // SampleJson
+    this.$jsonResponse = this.config.json_response;
+
+    // Sections
+    this.$detailsSection = this.$el.find('[data-dough-transition-details]');
+    this.$contributionsSection = this.$el.find('[data-dough-transition-contributions]');
+    this.$resultsSection = this.$el.find('[data-dough-transition-results]');
+
+    // Forms
+    this.$detailsForm = this.$el.find('[data-dough-details-form]');
+    this.$detailsFormPath = this.$detailsForm.attr('action');
+    this.$contributionForm = this.$el.find('[data-dough-contribution-form]');
+
+    // Headings
+    this.$detailsHeading = this.$el.find('.details__heading');
+
+    // Buttons
+    this.$detailsSubmit = this.$el.find('[data-dough-submit]');
+    this.$detailsEdit = this.$el.find('.js-edit-details');
+
+    // Fields
+    this.$ageField = this.$el.find('[data-dough-age-field]');
+    this.$genderField = this.$el.find('[data-dough-gender-select]');
+    this.$salaryField = this.$el.find('[data-dough-salary-input]');
+    this.$salaryFrequency = this.$el.find('[data-dough-frequency-select]');
+
+    // Options
+    this.$employerPartOption = this.$el.find('[data-dough-employer-part-radio]');
+    this.$employerFullOption = this.$el.find('[data-dough-employer-part-radio]');
+
+  };
+
+  /**
+  * Inherit from base module, for shared methods and interface
+  */
+  DoughBaseComponent.extend(StepTransition);
+
+  StepTransition.componentName = 'StepTransition';
+
+  StepTransition.prototype.init = function(initialised) {
+    this._initialisedSuccess(initialised);
+    this._setupEvents();
+  };
+
+  StepTransition.prototype._setupEvents = function() {
+    var $this = this;
+
+    $this.$detailsForm.submit(function(event){
+      event.preventDefault();
+      $this._prepareAJAXpost($this);
+    });
+  };
+
+  StepTransition.prototype._prepareAJAXpost = function($this) {
+
+    var postData = $this.$detailsForm.serialize();
+
+    $.ajax({
+      type: "post",
+      url: $this.$detailsFormPath,
+      data: postData,
+      contentType: "application/x-www-form-urlencoded",
+      success: function(responseData, textStatus, jqXHR) {
+        $this._receiveAJAXresponse(responseData);
+      },
+      error: function(jqXHR, textStatus, errorThrown) {
+        console.log(errorThrown);
+      }
+    })
+  };
+
+  StepTransition.prototype._receiveAJAXresponse = function(responseData) {
+    
+    var $this = this,
+        $eligibleSalary = responseData.eligible_salary,
+        $salaryFrequency = responseData.frequency,
+        $gender = responseData.gender,
+        $age = responseData.age,
+        $contribution = responseData.contribution;
```

Comment:
Sorry, seeing again, can we rename to contribution_preference, please? On the backend we are calling contribution_preference and it will be good to have the same names.